src/basic/random-util.c

```diff
@@ -248,3 +249,34 @@ uint64_t random_u64_range(uint64_t m) {
 
         return x % m;
 }
+
+/* Returns 1 if the seed file is okay, 0 if not, and < 0 on error determining. */
+int verify_random_seed_file_permissions(int seed_fd) {
+        _cleanup_free_ char *full_path = NULL, *prefix = NULL;
+        struct stat st;
+        int r;
+
+        if (fstat(seed_fd, &st) < 0)
+                return log_error_errno(errno, "Unable to fstat random seed file descriptor: %m");
+        if (!S_ISREG(st.st_mode))
+                return log_error_errno(EBADE, "Random seed file is not a regular file");
+        if (!(st.st_mode & 0007))
```

Comment:
doesn't matter much: we usually have the rule that we don't rely on C's downgrade-to-bool feature for stuff that isn't relaly boolean or boolean-like. and this isn't really, it can be anything in the range 0…7. Hence please do explicit numeric comparison.

```c
if ((st.st_mode & 0007) != 0)
``